### ISSUE TITLE
Coal veins does not spawn in the overworld

### ISSUE THREAD

**Issue by TWJedi** [Darkosto/SevTech-Ages#1707]
I first started my world in 3.0, and now in 3.0.6, coal veins do not spawn in newly-generated chunks. Coal samples spawn on the surface, but when I mine out the chunk, there are no veins. This has happened in five different chunks so far. 

Modpack version 3.0.6

**Comment by ElectroBot** [comment]
Are you running SP or on a server?

If SP or you are OP, then backup the world (just in case) and install https://minecraft.curseforge.com/projects/world-stripper in your client, then change the Mod Options to only 3 or 5 chunk radius.

Then strip your world.

**Comment by TheBrain0110** [comment]
Is it supposed to be not guaranteed that a sample indicates ore below? Because I've had 2 sets of coal samples I've found in different areas, and in one I found some coal no problem. 

The other I dug around and couldn't find it, and even after stripping away all the stone with commands, there definitely wasn't any in any of the chunks around the sample.

So either it didn't generate properly, or there is supposed to be some luck / randomness and a surface sample doesn't always mean there's anything to find below?

**Comment by artdude543** [comment]
I believe this in an RNG issue; on all the test worlds and from others playing the pack Ore Gen is pretty good (and some worlds too much). In regards to the chance, I don't believe that is a thing (I could be wrong) but I've found in my world some of the veins are in adjacent chunks.

You should be able to find the ore under each indicator as that's what they are for. So I can only suggest trying to mine around the chunks around the one the ore _should_ be in. But other than that we don't have enough evidence to suggest something is broken; as ore gen was boosted in the latest versions and the mod had a tonne of fixes for the generation. Which Darko tested heavily on stream before we pushed with the updated mod.

But if you can provide enough _soild_ evidence about the issue then please make a new issue with all the proof (and possible world file upload) so we can further look into the issue. But it could be just a case of bad luck sadly.

**Comment by Sorbe** [comment]
I have gotten the no actual coal seam present outcome now 3 times.  Twice there was nothing to be found either by fully scanning the chunk, laterally scanning while mining down to bedrock.   Coal surface indicators, not actual coal below the ground.  One time, I found exactly ONE coal block right at the surface and nothing more below that.  This last one block at the surface is suggestive of the actual bug's nature.

Allow me to speculate for a moment.  The supposed spawn depth for coal vein is very high compared to most ores.  It is high enough so that the chosen vein high could be above or at the surface land level.  I suspect that if the chosen spawn height is above grade, that those portions of the coal vein are clipped off.  And if it's right at grade, you'll only see a small number of block of coal at the surface.

Thus, I would speculate that the best way to fix this would be to extend the coal discover zone and lower the upper limit as well.    I will say that I have also found normal full coal seams at depth several times and the above problems about half the time.
